core/templates/pages/contributor-dashboard-admin-page/contributor-admin-dashboard-page.component.spec.ts

```diff
@@ -540,4 +544,16 @@ describe('Contributor dashboard Admin page', () => {
     expect(modalSpy).toHaveBeenCalledWith(UsernameInputModal);
     expect(openRoleEditorSpy).toHaveBeenCalledWith('user1');
   }));
+
+  it('should start any language with its English name', () => {
+    for (const language of component.languageChoices) {
+      if (language.id === 'hi-en') {
+        expect(language.language).toBe('Hinglish');
+      } else if (language.id === 'ms') {
+        expect(language.language).toBe('Bahasa Melayu (بهاس ملايو)');
+      } else if (language.id === 'az') {
+        expect(language.language).toBe('Azerbaijani (Azeri)');
+      }
+    }
```

Comment:
Generally speaking, if-else statements shouldn't really be used in top-level component tests. It might be better to create a Jasmine spy on `AppConstants.SUPPORTED_AUDIO_LANGUAGES` that returns our initial test data before arrangement.

Reply:
I am not sure how to do this. 
I tried this. 
spyOnProperty(AppConstants, 'SUPPORTED_AUDIO_LANGUAGES')
      .and.returnValue([
        {
          id: 'ms',
          description: 'بهاس ملايو(Bahasa Melayu)',
          relatedLanguages: ['ms'],
          direction: 'ltr'
        },
        ]);
     
But it is not working, it gives this [error](https://pastebin.com/SZqP2HGV).

Reply:
I think maybe what you can do is have something along the lines of "assert that languageChoices contains these three dicts". E.g. `expect(languageChoices.some(condition)).toBeTrue()` where condition checks that the id is ABC and the language is XYZ.

Reply:
Done. Used `toInclude`